app/modules/examinations/router_v2.py

```
@@ -83,6 +130,113 @@ def delete_examination(exam_id: int, db: Session = Depends(get_db), principal: P
+        "class_id": payload.class_id,
+        "teacher_id": payload.teacher_id,
+        "total_marks": payload.total_marks,
+    }
+
+
+@router.patch("/examinations/{exam_id}/subjects/{assignment_id}")
+def update_exam_subject_assignment(
+    exam_id: int,
+    assignment_id: int,
+    payload: ExamSubjectAssignment,
+    db: Session = Depends(get_db),
```

Ruff (0.16.1)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)